elastic/kibana
# Add summary capabilities to the API and execution logic

Resolves: #143376

This PR adds the alerts summary capability to the rule execution.
If an action is a summary action and not throttled (Summary of alerts on a custom interval) alerts summary is fetched from the AAD index and injected into the action message variables.

**To verify:**
Create a rule with an action and expect it to run as usual,
Then modify the action data in backend to mimic an action that has the new notifyWhen and throttle fields and expect it to run as expected as well.

I usually replace the following code with the below code snippet to mimic an action with the new fields:
https://github.com/elastic/kibana/blob/8f790d7d6a5c27d85a871e3203488192646fdbf2/x-pack/plugins/alerting/server/routes/create_rule.ts#L129

```
data: rewriteBodyReq({
   ...rule,
   actions: [
            {
                   group: rule.actions[0].group,
                   id: rule.actions[0].id,
                   params: rule.actions[0].params,
                   frequency: {
                         summary: true,
                          notifyWhen: 'onThrottleInterval',
                          throttle: '5m',
                   },
              },
   ],
   notify_when: undefined,
   throttle: undefined,
}),
```
